src/Compilers/CSharp/Portable/Binder/Binder_Conversions.cs

```diff
@@ -995,6 +974,61 @@ private BoundCollectionExpression ConvertCollectionExpression(
                     _ = GetWellKnownTypeMember(WellKnownMember.System_Collections_Generic_List_T__ToArray, diagnostics, syntax: syntax);
                 }
 
+                if (collectionTypeKind is CollectionExpressionTypeKind.CollectionBuilder)
```

Comment:
>if [](http://example.com/codeflow?start=16&length=2)

`else if`?

Reply:
We could, although the comment above the earlier `if` only applies to that `if { ... } else if { ... }` block, so to avoid confusion, I'll leave as separate `if`.